Collection Python requirements can no longer be installed on Mac ARM due to upstream uamqp dependency

##### SUMMARY
This is very closely related to https://github.com/ansible-collections/azure/issues/1505 but it is subtly different.

On up to date Mac systems a new build issue has appeared with the `uamqp` Python package, which is an upstream dependency of `azure-iot-hub`. There appears to be a guttural C build issue now with the latest versions on Mac that prevents building the package outright on any Python version.

Like the above issue the best course here is likely to remove [`azure-iot-hub`](https://github.com/Azure/azure-iot-hub-python/) as a requirement as it's a stale library that requires [`azure-uamqp-python`](https://github.com/Azure/azure-uamqp-python), which is no longer in active development.

An [issue](https://github.com/Azure/azure-uamqp-python/issues/386) on the [`azure-uamqp-python`](https://github.com/Azure/azure-uamqp-python) tracker has also been raised.

```
      clang -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include -I/opt/homebrew/opt/libpq/include -I./src/vendor/inc -I./src/vendor/azure-uamqp-c/deps/azure-macro-utils-c/inc -I./src/vendor/azure-uamqp-c/deps/umock-c/inc -I./src/vendor/azure-uamqp-c/deps/azure-c-shared-utility/pal/inc -I./src/vendor/azure-uamqp-c/deps/azure-c-shared-utility/inc -I./src/vendor/azure-uamqp-c/inc -I./src/vendor/azure-uamqp-c/deps/azure-c-shared-utility/pal/linux -I/Users/user/.asdf/installs/python/3.11.2/include/python3.11 -c uamqp/c_uamqp.c -o build/temp.macosx-13.2-arm64-cpython-311/uamqp/c_uamqp.o -g -O0 -std=gnu99 -fPIC
      uamqp/c_uamqp.c:10837:69: warning: incompatible pointer types passing 'PyObject *' (aka 'struct _object *') to parameter of type 'STRING_HANDLE' (aka 'struct STRING_TAG *') [-Wincompatible-pointer-types]
          __pyx_t_6 = ((STRING_concat_with_STRING(__pyx_v_self->_c_value, __pyx_t_1) != 0) != 0);
                                                                          ^~~~~~~~~
      ./src/vendor/azure-uamqp-c/deps/azure-c-shared-utility/inc/azure_c_shared_utility/strings.h:31:87: note: passing argument to parameter 's2' here
      MOCKABLE_FUNCTION(, int, STRING_concat_with_STRING, STRING_HANDLE, s1, STRING_HANDLE, s2);
                                                                                            ^
      uamqp/c_uamqp.c:17140:8: warning: passing 'const char *' to parameter of type 'void *' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
        free(__pyx_v_as_string);
             ^~~~~~~~~~~~~~~~~
      /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/malloc/_malloc.h:56:37: note: passing argument to parameter here
      void  free(void * __unsafe_indexable);
                                          ^
      uamqp/c_uamqp.c:18588:74: warning: incompatible pointer types passing 'int *' to parameter of type 'bool *' [-Wincompatible-pointer-types]
        __pyx_t_5 = ((amqpvalue_get_boolean(__pyx_v_self->__pyx_base._c_value, (&__pyx_v__value)) == 0) != 0);
                                                                               ^~~~~~~~~~~~~~~~~
      ./src/vendor/azure-uamqp-c/inc/azure_uamqp_c/amqpvalue.h:34:79: note: passing argument to parameter 'bool_value' here
          MOCKABLE_FUNCTION(, int, amqpvalue_get_boolean, AMQP_VALUE, value, bool*, bool_value);
                                                                                    ^
      uamqp/c_uamqp.c:41712:24: warning: cast to 'void *' from smaller integer type 'unsigned int' [-Wint-to-void-pointer-cast]
              __pyx_t_1 = ((((void *)__pyx_v_status_code) != NULL) != 0);
                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~
      uamqp/c_uamqp.c:44042:20: warning: cast to 'void *' from smaller integer type 'uint32_t' (aka 'unsigned int') [-Wint-to-void-pointer-cast]
          __pyx_t_1 = ((((void *)__pyx_v__value) == NULL) != 0);
                         ^~~~~~~~~~~~~~~~~~~~~~
      uamqp/c_uamqp.c:44303:20: warning: cast to 'void *' from smaller integer type 'uint16_t' (aka 'unsigned short') [-Wint-to-void-pointer-cast]
          __pyx_t_1 = ((((void *)__pyx_v__value) == NULL) != 0);
                         ^~~~~~~~~~~~~~~~~~~~~~
      uamqp/c_uamqp.c:44564:20: warning: cast to 'void *' from smaller integer type 'milliseconds' (aka 'unsigned int') [-Wint-to-void-pointer-cast]
          __pyx_t_1 = ((((void *)__pyx_v__value) == NULL) != 0);
                         ^~~~~~~~~~~~~~~~~~~~~~
      uamqp/c_uamqp.c:45081:20: warning: cast to 'void *' from smaller integer type 'uint32_t' (aka 'unsigned int') [-Wint-to-void-pointer-cast]
          __pyx_t_1 = ((((void *)__pyx_v__value) == NULL) != 0);
                         ^~~~~~~~~~~~~~~~~~~~~~
      uamqp/c_uamqp.c:48549:20: warning: cast to 'void *' from smaller integer type 'uint32_t' (aka 'unsigned int') [-Wint-to-void-pointer-cast]
          __pyx_t_1 = ((((void *)__pyx_v__value) == NULL) != 0);
                         ^~~~~~~~~~~~~~~~~~~~~~
      uamqp/c_uamqp.c:48797:20: warning: cast to 'void *' from smaller integer type 'milliseconds' (aka 'unsigned int') [-Wint-to-void-pointer-cast]
          __pyx_t_1 = ((((void *)__pyx_v__value) == NULL) != 0);
                         ^~~~~~~~~~~~~~~~~~~~~~
      uamqp/c_uamqp.c:49035:60: warning: incompatible pointer types passing 'int *' to parameter of type 'bool *' [-Wincompatible-pointer-types]
        __pyx_t_1 = ((header_get_durable(__pyx_v_self->_c_value, (&__pyx_v__value)) == 0) != 0);
                                                                 ^~~~~~~~~~~~~~~~~
      ./src/vendor/azure-uamqp-c/inc/azure_uamqp_c/amqp_definitions_header.h:33:80: note: passing argument to parameter 'durable_value' here
          MOCKABLE_FUNCTION(, int, header_get_durable, HEADER_HANDLE, header, bool*, durable_value);
                                                                                     ^
      uamqp/c_uamqp.c:49045:20: warning: cast to 'void *' from smaller integer type 'int' [-Wint-to-void-pointer-cast]
          __pyx_t_1 = ((((void *)__pyx_v__value) == NULL) != 0);
                         ^~~~~~~~~~~~~~~~~~~~~~
      uamqp/c_uamqp.c:49283:67: warning: incompatible pointer types passing 'int *' to parameter of type 'bool *' [-Wincompatible-pointer-types]
        __pyx_t_1 = ((header_get_first_acquirer(__pyx_v_self->_c_value, (&__pyx_v__value)) == 0) != 0);
                                                                        ^~~~~~~~~~~~~~~~~
      ./src/vendor/azure-uamqp-c/inc/azure_uamqp_c/amqp_definitions_header.h:39:87: note: passing argument to parameter 'first_acquirer_value' here
          MOCKABLE_FUNCTION(, int, header_get_first_acquirer, HEADER_HANDLE, header, bool*, first_acquirer_value);
                                                                                            ^
      uamqp/c_uamqp.c:49293:20: warning: cast to 'void *' from smaller integer type 'int' [-Wint-to-void-pointer-cast]
          __pyx_t_1 = ((((void *)__pyx_v__value) == NULL) != 0);
                         ^~~~~~~~~~~~~~~~~~~~~~
      uamqp/c_uamqp.c:49541:20: warning: cast to 'void *' from smaller integer type 'uint8_t' (aka 'unsigned char') [-Wint-to-void-pointer-cast]
          __pyx_t_1 = ((((void *)__pyx_v__value) == NULL) != 0);
                         ^~~~~~~~~~~~~~~~~~~~~~
      uamqp/c_uamqp.c:65557:146: error: incompatible function pointer types passing 'void (void *, enum MESSAGE_RECEIVER_STATE_TAG, enum MESSAGE_RECEIVER_STATE_TAG)' to parameter of type 'ON_MESSAGE_RECEIVER_STATE_CHANGED' (aka 'void (*)(const void *, enum MESSAGE_RECEIVER_STATE_TAG, enum MESSAGE_RECEIVER_STATE_TAG)') [-Wincompatible-function-pointer-types]
        __pyx_t_1 = ((struct __pyx_vtabstruct_5uamqp_7c_uamqp_cMessageReceiver *)__pyx_v_receiver->__pyx_vtab)->create(__pyx_v_receiver, __pyx_v_link, __pyx_f_5uamqp_7c_uamqp_on_message_receiver_state_changed, ((void *)__pyx_v_callback_context)); if (unlikely(!__pyx_t_1)) __PYX_ERR(7, 23, __pyx_L1_error)
                                                                                                                                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      uamqp/c_uamqp.c:71932:138: warning: passing 'const IO_INTERFACE_DESCRIPTION *' (aka 'const struct IO_INTERFACE_DESCRIPTION_TAG *') to parameter of type 'IO_INTERFACE_DESCRIPTION *' (aka 'struct IO_INTERFACE_DESCRIPTION_TAG *') discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
        __pyx_t_2 = ((struct __pyx_vtabstruct_5uamqp_7c_uamqp_IOInterfaceDescription *)__pyx_v_interface->__pyx_vtab)->wrap(__pyx_v_interface, __pyx_v_io_desc); if (unlikely(!__pyx_t_2)) __PYX_ERR(16, 50, __pyx_L1_error)
                                                                                                                                               ^~~~~~~~~~~~~~~
      uamqp/c_uamqp.c:74194:64: warning: passing 'char **' to parameter of type 'const char **' discards qualifiers in nested pointer types [-Wincompatible-pointer-types-discards-qualifiers]
        __pyx_t_1 = ((properties_get_subject(__pyx_v_self->_c_value, (&__pyx_v__value)) == 0) != 0);
                                                                     ^~~~~~~~~~~~~~~~~
      ./src/vendor/azure-uamqp-c/inc/azure_uamqp_c/amqp_definitions_properties.h:39:99: note: passing argument to parameter 'subject_value' here
          MOCKABLE_FUNCTION(, int, properties_get_subject, PROPERTIES_HANDLE, properties, const char**, subject_value);
                                                                                                        ^
      uamqp/c_uamqp.c:75046:69: warning: passing 'char **' to parameter of type 'const char **' discards qualifiers in nested pointer types [-Wincompatible-pointer-types-discards-qualifiers]
        __pyx_t_1 = ((properties_get_content_type(__pyx_v_self->_c_value, (&__pyx_v__value)) == 0) != 0);
                                                                          ^~~~~~~~~~~~~~~~~
      ./src/vendor/azure-uamqp-c/inc/azure_uamqp_c/amqp_definitions_properties.h:45:104: note: passing argument to parameter 'content_type_value' here
          MOCKABLE_FUNCTION(, int, properties_get_content_type, PROPERTIES_HANDLE, properties, const char**, content_type_value);
                                                                                                             ^
      uamqp/c_uamqp.c:75294:73: warning: passing 'char **' to parameter of type 'const char **' discards qualifiers in nested pointer types [-Wincompatible-pointer-types-discards-qualifiers]
        __pyx_t_1 = ((properties_get_content_encoding(__pyx_v_self->_c_value, (&__pyx_v__value)) == 0) != 0);
                                                                              ^~~~~~~~~~~~~~~~~
      ./src/vendor/azure-uamqp-c/inc/azure_uamqp_c/amqp_definitions_properties.h:47:108: note: passing argument to parameter 'content_encoding_value' here
          MOCKABLE_FUNCTION(, int, properties_get_content_encoding, PROPERTIES_HANDLE, properties, const char**, content_encoding_value);
                                                                                                                 ^
      uamqp/c_uamqp.c:76296:20: warning: cast to 'void *' from smaller integer type 'sequence_no' (aka 'unsigned int') [-Wint-to-void-pointer-cast]
          __pyx_t_1 = ((((void *)__pyx_v__value) == NULL) != 0);
                         ^~~~~~~~~~~~~~~~~~~~~~
      uamqp/c_uamqp.c:76534:74: warning: passing 'char **' to parameter of type 'const char **' discards qualifiers in nested pointer types [-Wincompatible-pointer-types-discards-qualifiers]
        __pyx_t_1 = ((properties_get_reply_to_group_id(__pyx_v_self->_c_value, (&__pyx_v__value)) == 0) != 0);
                                                                               ^~~~~~~~~~~~~~~~~
      ./src/vendor/azure-uamqp-c/inc/azure_uamqp_c/amqp_definitions_properties.h:57:109: note: passing argument to parameter 'reply_to_group_id_value' here
          MOCKABLE_FUNCTION(, int, properties_get_reply_to_group_id, PROPERTIES_HANDLE, properties, const char**, reply_to_group_id_value);
                                                                                                                  ^
      uamqp/c_uamqp.c:76934:139: warning: passing 'const SASL_MECHANISM_INTERFACE_DESCRIPTION *' (aka 'const struct SASL_MECHANISM_INTERFACE_TAG *') to parameter of type 'SASL_MECHANISM_INTERFACE_DESCRIPTION *' (aka 'struct SASL_MECHANISM_INTERFACE_TAG *') discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
        __pyx_t_2 = ((struct __pyx_vtabstruct_5uamqp_7c_uamqp_SASLMechanismInterfaceDescription *)__pyx_v_desc->__pyx_vtab)->wrap(__pyx_v_desc, __pyx_v_interface); if (unlikely(!__pyx_t_2)) __PYX_ERR(18, 24, __pyx_L1_error)
                                                                                                                                                ^~~~~~~~~~~~~~~~~
      uamqp/c_uamqp.c:77037:19: warning: assigning to 'SASL_MECHANISM_INTERFACE_DESCRIPTION *' (aka 'struct SASL_MECHANISM_INTERFACE_TAG *') from 'const SASL_MECHANISM_INTERFACE_DESCRIPTION *' (aka 'const struct SASL_MECHANISM_INTERFACE_TAG *') discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
        __pyx_v_io_desc = saslanonymous_get_interface();
                        ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      uamqp/c_uamqp.c:77193:19: warning: assigning to 'SASL_MECHANISM_INTERFACE_DESCRIPTION *' (aka 'struct SASL_MECHANISM_INTERFACE_TAG *') from 'const SASL_MECHANISM_INTERFACE_DESCRIPTION *' (aka 'const struct SASL_MECHANISM_INTERFACE_TAG *') discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
        __pyx_v_io_desc = saslplain_get_interface();
                        ^ ~~~~~~~~~~~~~~~~~~~~~~~~~
      uamqp/c_uamqp.c:83617:18: warning: cast to 'void *' from smaller integer type 'terminus_durability' (aka 'unsigned int') [-Wint-to-void-pointer-cast]
        __pyx_t_1 = ((((void *)__pyx_v__value) == NULL) != 0);
                       ^~~~~~~~~~~~~~~~~~~~~~
      uamqp/c_uamqp.c:84131:18: warning: cast to 'void *' from smaller integer type 'seconds' (aka 'unsigned int') [-Wint-to-void-pointer-cast]
        __pyx_t_1 = ((((void *)__pyx_v__value) == NULL) != 0);
                       ^~~~~~~~~~~~~~~~~~~~~~
      uamqp/c_uamqp.c:84343:60: warning: incompatible pointer types passing 'int *' to parameter of type 'bool *' [-Wincompatible-pointer-types]
        __pyx_t_1 = ((source_get_dynamic(__pyx_v_self->_c_value, (&__pyx_v__value)) != 0) != 0);
                                                                 ^~~~~~~~~~~~~~~~~
      ./src/vendor/azure-uamqp-c/inc/azure_uamqp_c/amqp_definitions_source.h:41:80: note: passing argument to parameter 'dynamic_value' here
          MOCKABLE_FUNCTION(, int, source_get_dynamic, SOURCE_HANDLE, source, bool*, dynamic_value);
                                                                                     ^
      uamqp/c_uamqp.c:84388:18: warning: cast to 'void *' from smaller integer type 'int' [-Wint-to-void-pointer-cast]
        __pyx_t_1 = ((((void *)__pyx_v__value) == NULL) != 0);
                       ^~~~~~~~~~~~~~~~~~~~~~
      uamqp/c_uamqp.c:86663:18: warning: cast to 'void *' from smaller integer type 'terminus_durability' (aka 'unsigned int') [-Wint-to-void-pointer-cast]
        __pyx_t_1 = ((((void *)__pyx_v__value) == NULL) != 0);
                       ^~~~~~~~~~~~~~~~~~~~~~
      uamqp/c_uamqp.c:87177:18: warning: cast to 'void *' from smaller integer type 'seconds' (aka 'unsigned int') [-Wint-to-void-pointer-cast]
        __pyx_t_1 = ((((void *)__pyx_v__value) == NULL) != 0);
                       ^~~~~~~~~~~~~~~~~~~~~~
      uamqp/c_uamqp.c:87389:60: warning: incompatible pointer types passing 'int *' to parameter of type 'bool *' [-Wincompatible-pointer-types]
        __pyx_t_1 = ((target_get_dynamic(__pyx_v_self->_c_value, (&__pyx_v__value)) != 0) != 0);
                                                                 ^~~~~~~~~~~~~~~~~
      ./src/vendor/azure-uamqp-c/inc/azure_uamqp_c/amqp_definitions_target.h:41:80: note: passing argument to parameter 'dynamic_value' here
          MOCKABLE_FUNCTION(, int, target_get_dynamic, TARGET_HANDLE, target, bool*, dynamic_value);
                                                                                     ^
      uamqp/c_uamqp.c:87434:18: warning: cast to 'void *' from smaller integer type 'int' [-Wint-to-void-pointer-cast]
        __pyx_t_1 = ((((void *)__pyx_v__value) == NULL) != 0);
                       ^~~~~~~~~~~~~~~~~~~~~~
      uamqp/c_uamqp.c:90457:109: warning: passing 'const IO_INTERFACE_DESCRIPTION *' (aka 'const struct IO_INTERFACE_DESCRIPTION_TAG *') to parameter of type 'IO_INTERFACE_DESCRIPTION *' (aka 'struct IO_INTERFACE_DESCRIPTION_TAG *') discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
        __pyx_t_1 = ((struct __pyx_vtabstruct_5uamqp_7c_uamqp_XIO *)__pyx_v_xio->__pyx_vtab)->create(__pyx_v_xio, __pyx_v_ws_io_interface, ((PyObject *)__pyx_v_io_config), (&__pyx_v_io_config->_c_value)); if (unlikely(!__pyx_t_1)) __PYX_ERR(24, 23, __pyx_L1_error)
                                                                                                                  ^~~~~~~~~~~~~~~~~~~~~~~
      uamqp/c_uamqp.c:90791:109: warning: passing 'const IO_INTERFACE_DESCRIPTION *' (aka 'const struct IO_INTERFACE_DESCRIPTION_TAG *') to parameter of type 'IO_INTERFACE_DESCRIPTION *' (aka 'struct IO_INTERFACE_DESCRIPTION_TAG *') discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
        __pyx_t_2 = ((struct __pyx_vtabstruct_5uamqp_7c_uamqp_XIO *)__pyx_v_xio->__pyx_vtab)->create(__pyx_v_xio, __pyx_v_interface, ((PyObject *)__pyx_v_io_config), (&__pyx_v_io_config->_c_value)); if (unlikely(!__pyx_t_2)) __PYX_ERR(24, 39, __pyx_L1_error)
                                                                                                                  ^~~~~~~~~~~~~~~~~
      uamqp/c_uamqp.c:111209:3: warning: unused label 'bad' [-Wunused-label]
        bad:
        ^~~~
      35 warnings and 1 error generated.
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]
```

I'm seeing this same issue building an Alpine linux x86_64 Docker image and installing Ansible (core 2.16, the yum/dnf tasks don't work on [rhel8 in Ansible core 2.17+](https://forum.ansible.com/t/issue-with-ansible-on-rocky-linux-8-10-python-3-12-future-feature-annotations-is-not-defined/41117/3)) with Pip and the collection with `ansible-galaxy collection install`. I've tried Alpine 3.19 (py 3.11), 3.20 (py 3.12), and 3.21 (py 3.12). I get this error with every version of the collection that I've tried (basically >= 2.0, since Ansible core 2.16 includes v1.19.0 of the collection). I'm guessing there's something simple I've missed or I would expect to see this being reported much more heavily than the ARM issue has been.

```shell
72.24   Building wheel for uamqp (pyproject.toml): started
73.26   Building wheel for uamqp (pyproject.toml): finished with status 'error'
73.26   error: subprocess-exited-with-error
73.26   
73.26   × Building wheel for uamqp (pyproject.toml) did not run successfully.
73.26   │ exit code: 1
73.26   ╰─> [148 lines of output]
73.26       /tmp/pip-build-env-rctzd9vh/overlay/lib/python3.12/site-packages/setuptools/dist.py:759: SetuptoolsDeprecationWarning: License classifiers are deprecated.
73.26       !!
73.26       
73.26               ********************************************************************************
73.26               Please consider removing the following classifiers in favor of a SPDX license expression:
73.26       
73.26               License :: OSI Approved :: MIT License
73.26       
73.26               See https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#license for details.
73.26               ********************************************************************************
73.26       
73.26       !!
73.26         self._finalize_license_expression()
73.26       running bdist_wheel
73.26       running build
73.26       running build_py
73.26       creating build/lib.linux-x86_64-cpython-312/uamqp
73.26       copying uamqp/mgmt_operation.py -> build/lib.linux-x86_64-cpython-312/uamqp
73.26       copying uamqp/client.py -> build/lib.linux-x86_64-cpython-312/uamqp
73.26       copying uamqp/sender.py -> build/lib.linux-x86_64-cpython-312/uamqp
73.26       copying uamqp/__init__.py -> build/lib.linux-x86_64-cpython-312/uamqp
73.26       copying uamqp/address.py -> build/lib.linux-x86_64-cpython-312/uamqp
73.26       copying uamqp/types.py -> build/lib.linux-x86_64-cpython-312/uamqp
73.26       copying uamqp/connection.py -> build/lib.linux-x86_64-cpython-312/uamqp
73.26       copying uamqp/receiver.py -> build/lib.linux-x86_64-cpython-312/uamqp
73.26       copying uamqp/constants.py -> build/lib.linux-x86_64-cpython-312/uamqp
73.26       copying uamqp/utils.py -> build/lib.linux-x86_64-cpython-312/uamqp
73.26       copying uamqp/errors.py -> build/lib.linux-x86_64-cpython-312/uamqp
73.26       copying uamqp/message.py -> build/lib.linux-x86_64-cpython-312/uamqp
73.26       copying uamqp/session.py -> build/lib.linux-x86_64-cpython-312/uamqp
73.26       copying uamqp/compat.py -> build/lib.linux-x86_64-cpython-312/uamqp
73.26       creating build/lib.linux-x86_64-cpython-312/uamqp/authentication
73.26       copying uamqp/authentication/__init__.py -> build/lib.linux-x86_64-cpython-312/uamqp/authentication
73.26       copying uamqp/authentication/common.py -> build/lib.linux-x86_64-cpython-312/uamqp/authentication
73.26       copying uamqp/authentication/cbs_auth.py -> build/lib.linux-x86_64-cpython-312/uamqp/authentication
73.26       copying uamqp/authentication/cbs_auth_async.py -> build/lib.linux-x86_64-cpython-312/uamqp/authentication
73.26       creating build/lib.linux-x86_64-cpython-312/uamqp/async_ops
73.26       copying uamqp/async_ops/sender_async.py -> build/lib.linux-x86_64-cpython-312/uamqp/async_ops
73.26       copying uamqp/async_ops/mgmt_operation_async.py -> build/lib.linux-x86_64-cpython-312/uamqp/async_ops
73.26       copying uamqp/async_ops/client_async.py -> build/lib.linux-x86_64-cpython-312/uamqp/async_ops
73.26       copying uamqp/async_ops/connection_async.py -> build/lib.linux-x86_64-cpython-312/uamqp/async_ops
73.26       copying uamqp/async_ops/__init__.py -> build/lib.linux-x86_64-cpython-312/uamqp/async_ops
73.26       copying uamqp/async_ops/session_async.py -> build/lib.linux-x86_64-cpython-312/uamqp/async_ops
73.26       copying uamqp/async_ops/utils.py -> build/lib.linux-x86_64-cpython-312/uamqp/async_ops
73.26       copying uamqp/async_ops/receiver_async.py -> build/lib.linux-x86_64-cpython-312/uamqp/async_ops
73.26       running egg_info
73.26       writing uamqp.egg-info/PKG-INFO
73.26       writing dependency_links to uamqp.egg-info/dependency_links.txt
73.26       writing requirements to uamqp.egg-info/requires.txt
73.26       writing top-level names to uamqp.egg-info/top_level.txt
73.26       reading manifest file 'uamqp.egg-info/SOURCES.txt'
73.26       reading manifest template 'MANIFEST.in'
73.26       adding license file 'LICENSE'
73.26       writing manifest file 'uamqp.egg-info/SOURCES.txt'
73.26       copying uamqp/c_uamqp.c -> build/lib.linux-x86_64-cpython-312/uamqp
73.26       running build_ext
73.26       will build uamqp in build/temp.linux-x86_64-cpython-312/cmake
73.26       Building with generator flags: -G "Unix Makefiles"
73.26       calling cmake /tmp/pip-install-c2wxz4yn/uamqp_bc79c26f4cca49e8933ec96fa6cb39d9/src/vendor/azure-uamqp-c/ -G "Unix Makefiles" -Duse_openssl:bool=ON -Duse_default_uuid:bool=ON  -Duse_builtin_httpapi:bool=ON  -Dskip_samples:bool=ON -DCMAKE_POSITION_INDEPENDENT_CODE=TRUE -DCMAKE_BUILD_TYPE=Release
73.26       CMake Deprecation Warning at CMakeLists.txt:4 (cmake_minimum_required):
73.26         Compatibility with CMake < 3.10 will be removed from a future version of
73.26         CMake.
73.26       
73.26         Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
73.26         to tell CMake that the project requires at least <min> but has been updated
73.26         to work with policies introduced by <max> or earlier.
73.26       
73.26       
73.26       -- The C compiler identification is GNU 14.2.0
73.26       -- The CXX compiler identification is GNU 14.2.0
73.26       -- Detecting C compiler ABI info
73.26       -- Detecting C compiler ABI info - done
73.26       -- Check for working C compiler: /usr/bin/cc - skipped
73.26       -- Detecting C compile features
73.26       -- Detecting C compile features - done
73.26       -- Detecting CXX compiler ABI info
73.26       -- Detecting CXX compiler ABI info - done
73.26       -- Check for working CXX compiler: /usr/bin/c++ - skipped
73.26       -- Detecting CXX compile features
73.26       -- Detecting CXX compile features - done
73.26       CMake Error at deps/azure-macro-utils-c/CMakeLists.txt:4 (cmake_minimum_required):
73.26         Compatibility with CMake < 3.5 has been removed from CMake.
73.26       
73.26         Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
73.26         to tell CMake that the project requires at least <min> but has been updated
73.26         to work with policies introduced by <max> or earlier.
73.26       
73.26         Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.
73.26       
73.26       
73.26       -- Configuring incomplete, errors occurred!
73.26       Traceback (most recent call last):
73.26         File "/usr/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
73.26           main()
73.26         File "/usr/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
73.26           json_out['return_val'] = hook(**hook_input['kwargs'])
73.26                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
73.26         File "/usr/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 251, in build_wheel
73.26           return _build_backend().build_wheel(wheel_directory, config_settings,
73.26                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
73.26         File "/tmp/pip-build-env-rctzd9vh/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 435, in build_wheel
73.26           return _build(['bdist_wheel', '--dist-info-dir', str(metadata_directory)])
73.26                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
73.26         File "/tmp/pip-build-env-rctzd9vh/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 423, in _build
73.26           return self._build_with_temp_dir(
73.26                  ^^^^^^^^^^^^^^^^^^^^^^^^^^
73.26         File "/tmp/pip-build-env-rctzd9vh/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 404, in _build_with_temp_dir
73.26           self.run_setup()
73.26         File "/tmp/pip-build-env-rctzd9vh/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 512, in run_setup
73.26           super().run_setup(setup_script=setup_script)
73.26         File "/tmp/pip-build-env-rctzd9vh/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 317, in run_setup
73.26           exec(code, locals())
73.26         File "<string>", line 260, in <module>
73.26         File "/tmp/pip-build-env-rctzd9vh/overlay/lib/python3.12/site-packages/setuptools/__init__.py", line 117, in setup
73.26           return distutils.core.setup(**attrs)
73.26                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
73.26         File "/tmp/pip-build-env-rctzd9vh/overlay/lib/python3.12/site-packages/setuptools/_distutils/core.py", line 186, in setup
73.26           return run_commands(dist)
73.26                  ^^^^^^^^^^^^^^^^^^
73.26         File "/tmp/pip-build-env-rctzd9vh/overlay/lib/python3.12/site-packages/setuptools/_distutils/core.py", line 202, in run_commands
73.26           dist.run_commands()
73.26         File "/tmp/pip-build-env-rctzd9vh/overlay/lib/python3.12/site-packages/setuptools/_distutils/dist.py", line 1002, in run_commands
73.26           self.run_command(cmd)
73.26         File "/tmp/pip-build-env-rctzd9vh/overlay/lib/python3.12/site-packages/setuptools/dist.py", line 1104, in run_command
73.26           super().run_command(command)
73.26         File "/tmp/pip-build-env-rctzd9vh/overlay/lib/python3.12/site-packages/setuptools/_distutils/dist.py", line 1021, in run_command
73.26           cmd_obj.run()
73.26         File "/tmp/pip-build-env-rctzd9vh/overlay/lib/python3.12/site-packages/setuptools/command/bdist_wheel.py", line 370, in run
73.26           self.run_command("build")
73.26         File "/tmp/pip-build-env-rctzd9vh/overlay/lib/python3.12/site-packages/setuptools/_distutils/cmd.py", line 357, in run_command
73.26           self.distribution.run_command(command)
73.26         File "/tmp/pip-build-env-rctzd9vh/overlay/lib/python3.12/site-packages/setuptools/dist.py", line 1104, in run_command
73.26           super().run_command(command)
73.26         File "/tmp/pip-build-env-rctzd9vh/overlay/lib/python3.12/site-packages/setuptools/_distutils/dist.py", line 1021, in run_command
73.26           cmd_obj.run()
73.26         File "/tmp/pip-build-env-rctzd9vh/overlay/lib/python3.12/site-packages/setuptools/_distutils/command/build.py", line 135, in run
73.26           self.run_command(cmd_name)
73.26         File "/tmp/pip-build-env-rctzd9vh/overlay/lib/python3.12/site-packages/setuptools/_distutils/cmd.py", line 357, in run_command
73.26           self.distribution.run_command(command)
73.26         File "/tmp/pip-build-env-rctzd9vh/overlay/lib/python3.12/site-packages/setuptools/dist.py", line 1104, in run_command
73.26           super().run_command(command)
73.26         File "/tmp/pip-build-env-rctzd9vh/overlay/lib/python3.12/site-packages/setuptools/_distutils/dist.py", line 1021, in run_command
73.26           cmd_obj.run()
73.26         File "<string>", line 134, in run
73.26         File "<string>", line 183, in build_cmake
73.26         File "/usr/lib/python3.12/subprocess.py", line 413, in check_call
73.26           raise CalledProcessError(retcode, cmd)
73.26       subprocess.CalledProcessError: Command 'cmake /tmp/pip-install-c2wxz4yn/uamqp_bc79c26f4cca49e8933ec96fa6cb39d9/src/vendor/azure-uamqp-c/ -G "Unix Makefiles" -Duse_openssl:bool=ON -Duse_default_uuid:bool=ON  -Duse_builtin_httpapi:bool=ON  -Dskip_samples:bool=ON -DCMAKE_POSITION_INDEPENDENT_CODE=TRUE -DCMAKE_BUILD_TYPE=Release' returned non-zero exit status 1.
73.26       [end of output]
73.26   
73.26   note: This error originates from a subprocess, and is likely not a problem with pip.
73.26   ERROR: Failed building wheel for uamqp
```

> ##### SUMMARY
> This is very closely related to [#1505](https://github.com/ansible-collections/azure/issues/1505) but it is subtly different.
> 
> On up to date Mac systems a new build issue has appeared with the `uamqp` Python package, which is an upstream dependency of `azure-iot-hub`. There appears to be a guttural C build issue now with the latest versions on Mac that prevents building the package outright on any Python version.
> 
> Like the above issue the best course here is likely to remove [`azure-iot-hub`](https://github.com/Azure/azure-iot-hub-python/) as a requirement as it's a stale library that requires [`azure-uamqp-python`](https://github.com/Azure/azure-uamqp-python), which is no longer in active development.
> 
> An [issue](https://github.com/Azure/azure-uamqp-python/issues/386) on the [`azure-uamqp-python`](https://github.com/Azure/azure-uamqp-python) tracker has also been raised.
> 
> ```
>       clang -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include -I/opt/homebrew/opt/libpq/include -I./src/vendor/inc -I./src/vendor/azure-uamqp-c/deps/azure-macro-utils-c/inc -I./src/vendor/azure-uamqp-c/deps/umock-c/inc -I./src/vendor/azure-uamqp-c/deps/azure-c-shared-utility/pal/inc -I./src/vendor/azure-uamqp-c/deps/azure-c-shared-utility/inc -I./src/vendor/azure-uamqp-c/inc -I./src/vendor/azure-uamqp-c/deps/azure-c-shared-utility/pal/linux -I/Users/user/.asdf/installs/python/3.11.2/include/python3.11 -c uamqp/c_uamqp.c -o build/temp.macosx-13.2-arm64-cpython-311/uamqp/c_uamqp.o -g -O0 -std=gnu99 -fPIC
>       uamqp/c_uamqp.c:10837:69: warning: incompatible pointer types passing 'PyObject *' (aka 'struct _object *') to parameter of type 'STRING_HANDLE' (aka 'struct STRING_TAG *') [-Wincompatible-pointer-types]
>           __pyx_t_6 = ((STRING_concat_with_STRING(__pyx_v_self->_c_value, __pyx_t_1) != 0) != 0);
>                                                                           ^~~~~~~~~
>       ./src/vendor/azure-uamqp-c/deps/azure-c-shared-utility/inc/azure_c_shared_utility/strings.h:31:87: note: passing argument to parameter 's2' here
>       MOCKABLE_FUNCTION(, int, STRING_concat_with_STRING, STRING_HANDLE, s1, STRING_HANDLE, s2);
>                                                                                             ^
>       uamqp/c_uamqp.c:17140:8: warning: passing 'const char *' to parameter of type 'void *' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
>         free(__pyx_v_as_string);
>              ^~~~~~~~~~~~~~~~~
>       /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/malloc/_malloc.h:56:37: note: passing argument to parameter here
>       void  free(void * __unsafe_indexable);
>                                           ^
>       uamqp/c_uamqp.c:18588:74: warning: incompatible pointer types passing 'int *' to parameter of type 'bool *' [-Wincompatible-pointer-types]
>         __pyx_t_5 = ((amqpvalue_get_boolean(__pyx_v_self->__pyx_base._c_value, (&__pyx_v__value)) == 0) != 0);
>                                                                                ^~~~~~~~~~~~~~~~~
>       ./src/vendor/azure-uamqp-c/inc/azure_uamqp_c/amqpvalue.h:34:79: note: passing argument to parameter 'bool_value' here
>           MOCKABLE_FUNCTION(, int, amqpvalue_get_boolean, AMQP_VALUE, value, bool*, bool_value);
>                                                                                     ^
>       uamqp/c_uamqp.c:41712:24: warning: cast to 'void *' from smaller integer type 'unsigned int' [-Wint-to-void-pointer-cast]
>               __pyx_t_1 = ((((void *)__pyx_v_status_code) != NULL) != 0);
>                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~
>       uamqp/c_uamqp.c:44042:20: warning: cast to 'void *' from smaller integer type 'uint32_t' (aka 'unsigned int') [-Wint-to-void-pointer-cast]
>           __pyx_t_1 = ((((void *)__pyx_v__value) == NULL) != 0);
>                          ^~~~~~~~~~~~~~~~~~~~~~
>       uamqp/c_uamqp.c:44303:20: warning: cast to 'void *' from smaller integer type 'uint16_t' (aka 'unsigned short') [-Wint-to-void-pointer-cast]
>           __pyx_t_1 = ((((void *)__pyx_v__value) == NULL) != 0);
>                          ^~~~~~~~~~~~~~~~~~~~~~
>       uamqp/c_uamqp.c:44564:20: warning: cast to 'void *' from smaller integer type 'milliseconds' (aka 'unsigned int') [-Wint-to-void-pointer-cast]
>           __pyx_t_1 = ((((void *)__pyx_v__value) == NULL) != 0);
>                          ^~~~~~~~~~~~~~~~~~~~~~
>       uamqp/c_uamqp.c:45081:20: warning: cast to 'void *' from smaller integer type 'uint32_t' (aka 'unsigned int') [-Wint-to-void-pointer-cast]
>           __pyx_t_1 = ((((void *)__pyx_v__value) == NULL) != 0);
>                          ^~~~~~~~~~~~~~~~~~~~~~
>       uamqp/c_uamqp.c:48549:20: warning: cast to 'void *' from smaller integer type 'uint32_t' (aka 'unsigned int') [-Wint-to-void-pointer-cast]
>           __pyx_t_1 = ((((void *)__pyx_v__value) == NULL) != 0);
>                          ^~~~~~~~~~~~~~~~~~~~~~
>       uamqp/c_uamqp.c:48797:20: warning: cast to 'void *' from smaller integer type 'milliseconds' (aka 'unsigned int') [-Wint-to-void-pointer-cast]
>           __pyx_t_1 = ((((void *)__pyx_v__value) == NULL) != 0);
>                          ^~~~~~~~~~~~~~~~~~~~~~
>       uamqp/c_uamqp.c:49035:60: warning: incompatible pointer types passing 'int *' to parameter of type 'bool *' [-Wincompatible-pointer-types]
>         __pyx_t_1 = ((header_get_durable(__pyx_v_self->_c_value, (&__pyx_v__value)) == 0) != 0);
>                                                                  ^~~~~~~~~~~~~~~~~
>       ./src/vendor/azure-uamqp-c/inc/azure_uamqp_c/amqp_definitions_header.h:33:80: note: passing argument to parameter 'durable_value' here
>           MOCKABLE_FUNCTION(, int, header_get_durable, HEADER_HANDLE, header, bool*, durable_value);
>                                                                                      ^
>       uamqp/c_uamqp.c:49045:20: warning: cast to 'void *' from smaller integer type 'int' [-Wint-to-void-pointer-cast]
>           __pyx_t_1 = ((((void *)__pyx_v__value) == NULL) != 0);
>                          ^~~~~~~~~~~~~~~~~~~~~~
>       uamqp/c_uamqp.c:49283:67: warning: incompatible pointer types passing 'int *' to parameter of type 'bool *' [-Wincompatible-pointer-types]
>         __pyx_t_1 = ((header_get_first_acquirer(__pyx_v_self->_c_value, (&__pyx_v__value)) == 0) != 0);
>                                                                         ^~~~~~~~~~~~~~~~~
>       ./src/vendor/azure-uamqp-c/inc/azure_uamqp_c/amqp_definitions_header.h:39:87: note: passing argument to parameter 'first_acquirer_value' here
>           MOCKABLE_FUNCTION(, int, header_get_first_acquirer, HEADER_HANDLE, header, bool*, first_acquirer_value);
>                                                                                             ^
>       uamqp/c_uamqp.c:49293:20: warning: cast to 'void *' from smaller integer type 'int' [-Wint-to-void-pointer-cast]
>           __pyx_t_1 = ((((void *)__pyx_v__value) == NULL) != 0);
>                          ^~~~~~~~~~~~~~~~~~~~~~
>       uamqp/c_uamqp.c:49541:20: warning: cast to 'void *' from smaller integer type 'uint8_t' (aka 'unsigned char') [-Wint-to-void-pointer-cast]
>           __pyx_t_1 = ((((void *)__pyx_v__value) == NULL) != 0);
>                          ^~~~~~~~~~~~~~~~~~~~~~
>       uamqp/c_uamqp.c:65557:146: error: incompatible function pointer types passing 'void (void *, enum MESSAGE_RECEIVER_STATE_TAG, enum MESSAGE_RECEIVER_STATE_TAG)' to parameter of type 'ON_MESSAGE_RECEIVER_STATE_CHANGED' (aka 'void (*)(const void *, enum MESSAGE_RECEIVER_STATE_TAG, enum MESSAGE_RECEIVER_STATE_TAG)') [-Wincompatible-function-pointer-types]
>         __pyx_t_1 = ((struct __pyx_vtabstruct_5uamqp_7c_uamqp_cMessageReceiver *)__pyx_v_receiver->__pyx_vtab)->create(__pyx_v_receiver, __pyx_v_link, __pyx_f_5uamqp_7c_uamqp_on_message_receiver_state_changed, ((void *)__pyx_v_callback_context)); if (unlikely(!__pyx_t_1)) __PYX_ERR(7, 23, __pyx_L1_error)
>                                                                                                                                                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>       uamqp/c_uamqp.c:71932:138: warning: passing 'const IO_INTERFACE_DESCRIPTION *' (aka 'const struct IO_INTERFACE_DESCRIPTION_TAG *') to parameter of type 'IO_INTERFACE_DESCRIPTION *' (aka 'struct IO_INTERFACE_DESCRIPTION_TAG *') discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
>         __pyx_t_2 = ((struct __pyx_vtabstruct_5uamqp_7c_uamqp_IOInterfaceDescription *)__pyx_v_interface->__pyx_vtab)->wrap(__pyx_v_interface, __pyx_v_io_desc); if (unlikely(!__pyx_t_2)) __PYX_ERR(16, 50, __pyx_L1_error)
>                                                                                                                                                ^~~~~~~~~~~~~~~
>       uamqp/c_uamqp.c:74194:64: warning: passing 'char **' to parameter of type 'const char **' discards qualifiers in nested pointer types [-Wincompatible-pointer-types-discards-qualifiers]
>         __pyx_t_1 = ((properties_get_subject(__pyx_v_self->_c_value, (&__pyx_v__value)) == 0) != 0);
>                                                                      ^~~~~~~~~~~~~~~~~
>       ./src/vendor/azure-uamqp-c/inc/azure_uamqp_c/amqp_definitions_properties.h:39:99: note: passing argument to parameter 'subject_value' here
>           MOCKABLE_FUNCTION(, int, properties_get_subject, PROPERTIES_HANDLE, properties, const char**, subject_value);
>                                                                                                         ^
>       uamqp/c_uamqp.c:75046:69: warning: passing 'char **' to parameter of type 'const char **' discards qualifiers in nested pointer types [-Wincompatible-pointer-types-discards-qualifiers]
>         __pyx_t_1 = ((properties_get_content_type(__pyx_v_self->_c_value, (&__pyx_v__value)) == 0) != 0);
>                                                                           ^~~~~~~~~~~~~~~~~
>       ./src/vendor/azure-uamqp-c/inc/azure_uamqp_c/amqp_definitions_properties.h:45:104: note: passing argument to parameter 'content_type_value' here
>           MOCKABLE_FUNCTION(, int, properties_get_content_type, PROPERTIES_HANDLE, properties, const char**, content_type_value);
>                                                                                                              ^
>       uamqp/c_uamqp.c:75294:73: warning: passing 'char **' to parameter of type 'const char **' discards qualifiers in nested pointer types [-Wincompatible-pointer-types-discards-qualifiers]
>         __pyx_t_1 = ((properties_get_content_encoding(__pyx_v_self->_c_value, (&__pyx_v__value)) == 0) != 0);
>                                                                               ^~~~~~~~~~~~~~~~~
>       ./src/vendor/azure-uamqp-c/inc/azure_uamqp_c/amqp_definitions_properties.h:47:108: note: passing argument to parameter 'content_encoding_value' here
>           MOCKABLE_FUNCTION(, int, properties_get_content_encoding, PROPERTIES_HANDLE, properties, const char**, content_encoding_value);
>                                                                                                                  ^
>       uamqp/c_uamqp.c:76296:20: warning: cast to 'void *' from smaller integer type 'sequence_no' (aka 'unsigned int') [-Wint-to-void-pointer-cast]
>           __pyx_t_1 = ((((void *)__pyx_v__value) == NULL) != 0);
>                          ^~~~~~~~~~~~~~~~~~~~~~
>       uamqp/c_uamqp.c:76534:74: warning: passing 'char **' to parameter of type 'const char **' discards qualifiers in nested pointer types [-Wincompatible-pointer-types-discards-qualifiers]
>         __pyx_t_1 = ((properties_get_reply_to_group_id(__pyx_v_self->_c_value, (&__pyx_v__value)) == 0) != 0);
>                                                                                ^~~~~~~~~~~~~~~~~
>       ./src/vendor/azure-uamqp-c/inc/azure_uamqp_c/amqp_definitions_properties.h:57:109: note: passing argument to parameter 'reply_to_group_id_value' here
>           MOCKABLE_FUNCTION(, int, properties_get_reply_to_group_id, PROPERTIES_HANDLE, properties, const char**, reply_to_group_id_value);
>                                                                                                                   ^
>       uamqp/c_uamqp.c:76934:139: warning: passing 'const SASL_MECHANISM_INTERFACE_DESCRIPTION *' (aka 'const struct SASL_MECHANISM_INTERFACE_TAG *') to parameter of type 'SASL_MECHANISM_INTERFACE_DESCRIPTION *' (aka 'struct SASL_MECHANISM_INTERFACE_TAG *') discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
>         __pyx_t_2 = ((struct __pyx_vtabstruct_5uamqp_7c_uamqp_SASLMechanismInterfaceDescription *)__pyx_v_desc->__pyx_vtab)->wrap(__pyx_v_desc, __pyx_v_interface); if (unlikely(!__pyx_t_2)) __PYX_ERR(18, 24, __pyx_L1_error)
>                                                                                                                                                 ^~~~~~~~~~~~~~~~~
>       uamqp/c_uamqp.c:77037:19: warning: assigning to 'SASL_MECHANISM_INTERFACE_DESCRIPTION *' (aka 'struct SASL_MECHANISM_INTERFACE_TAG *') from 'const SASL_MECHANISM_INTERFACE_DESCRIPTION *' (aka 'const struct SASL_MECHANISM_INTERFACE_TAG *') discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
>         __pyx_v_io_desc = saslanonymous_get_interface();
>                         ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>       uamqp/c_uamqp.c:77193:19: warning: assigning to 'SASL_MECHANISM_INTERFACE_DESCRIPTION *' (aka 'struct SASL_MECHANISM_INTERFACE_TAG *') from 'const SASL_MECHANISM_INTERFACE_DESCRIPTION *' (aka 'const struct SASL_MECHANISM_INTERFACE_TAG *') discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
>         __pyx_v_io_desc = saslplain_get_interface();
>                         ^ ~~~~~~~~~~~~~~~~~~~~~~~~~
>       uamqp/c_uamqp.c:83617:18: warning: cast to 'void *' from smaller integer type 'terminus_durability' (aka 'unsigned int') [-Wint-to-void-pointer-cast]
>         __pyx_t_1 = ((((void *)__pyx_v__value) == NULL) != 0);
>                        ^~~~~~~~~~~~~~~~~~~~~~
>       uamqp/c_uamqp.c:84131:18: warning: cast to 'void *' from smaller integer type 'seconds' (aka 'unsigned int') [-Wint-to-void-pointer-cast]
>         __pyx_t_1 = ((((void *)__pyx_v__value) == NULL) != 0);
>                        ^~~~~~~~~~~~~~~~~~~~~~
>       uamqp/c_uamqp.c:84343:60: warning: incompatible pointer types passing 'int *' to parameter of type 'bool *' [-Wincompatible-pointer-types]
>         __pyx_t_1 = ((source_get_dynamic(__pyx_v_self->_c_value, (&__pyx_v__value)) != 0) != 0);
>                                                                  ^~~~~~~~~~~~~~~~~
>       ./src/vendor/azure-uamqp-c/inc/azure_uamqp_c/amqp_definitions_source.h:41:80: note: passing argument to parameter 'dynamic_value' here
>           MOCKABLE_FUNCTION(, int, source_get_dynamic, SOURCE_HANDLE, source, bool*, dynamic_value);
>                                                                                      ^
>       uamqp/c_uamqp.c:84388:18: warning: cast to 'void *' from smaller integer type 'int' [-Wint-to-void-pointer-cast]
>         __pyx_t_1 = ((((void *)__pyx_v__value) == NULL) != 0);
>                        ^~~~~~~~~~~~~~~~~~~~~~
>       uamqp/c_uamqp.c:86663:18: warning: cast to 'void *' from smaller integer type 'terminus_durability' (aka 'unsigned int') [-Wint-to-void-pointer-cast]
>         __pyx_t_1 = ((((void *)__pyx_v__value) == NULL) != 0);
>                        ^~~~~~~~~~~~~~~~~~~~~~
>       uamqp/c_uamqp.c:87177:18: warning: cast to 'void *' from smaller integer type 'seconds' (aka 'unsigned int') [-Wint-to-void-pointer-cast]
>         __pyx_t_1 = ((((void *)__pyx_v__value) == NULL) != 0);
>                        ^~~~~~~~~~~~~~~~~~~~~~
>       uamqp/c_uamqp.c:87389:60: warning: incompatible pointer types passing 'int *' to parameter of type 'bool *' [-Wincompatible-pointer-types]
>         __pyx_t_1 = ((target_get_dynamic(__pyx_v_self->_c_value, (&__pyx_v__value)) != 0) != 0);
>                                                                  ^~~~~~~~~~~~~~~~~
>       ./src/vendor/azure-uamqp-c/inc/azure_uamqp_c/amqp_definitions_target.h:41:80: note: passing argument to parameter 'dynamic_value' here
>           MOCKABLE_FUNCTION(, int, target_get_dynamic, TARGET_HANDLE, target, bool*, dynamic_value);
>                                                                                      ^
>       uamqp/c_uamqp.c:87434:18: warning: cast to 'void *' from smaller integer type 'int' [-Wint-to-void-pointer-cast]
>         __pyx_t_1 = ((((void *)__pyx_v__value) == NULL) != 0);
>                        ^~~~~~~~~~~~~~~~~~~~~~
>       uamqp/c_uamqp.c:90457:109: warning: passing 'const IO_INTERFACE_DESCRIPTION *' (aka 'const struct IO_INTERFACE_DESCRIPTION_TAG *') to parameter of type 'IO_INTERFACE_DESCRIPTION *' (aka 'struct IO_INTERFACE_DESCRIPTION_TAG *') discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
>         __pyx_t_1 = ((struct __pyx_vtabstruct_5uamqp_7c_uamqp_XIO *)__pyx_v_xio->__pyx_vtab)->create(__pyx_v_xio, __pyx_v_ws_io_interface, ((PyObject *)__pyx_v_io_config), (&__pyx_v_io_config->_c_value)); if (unlikely(!__pyx_t_1)) __PYX_ERR(24, 23, __pyx_L1_error)
>                                                                                                                   ^~~~~~~~~~~~~~~~~~~~~~~
>       uamqp/c_uamqp.c:90791:109: warning: passing 'const IO_INTERFACE_DESCRIPTION *' (aka 'const struct IO_INTERFACE_DESCRIPTION_TAG *') to parameter of type 'IO_INTERFACE_DESCRIPTION *' (aka 'struct IO_INTERFACE_DESCRIPTION_TAG *') discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
>         __pyx_t_2 = ((struct __pyx_vtabstruct_5uamqp_7c_uamqp_XIO *)__pyx_v_xio->__pyx_vtab)->create(__pyx_v_xio, __pyx_v_interface, ((PyObject *)__pyx_v_io_config), (&__pyx_v_io_config->_c_value)); if (unlikely(!__pyx_t_2)) __PYX_ERR(24, 39, __pyx_L1_error)
>                                                                                                                   ^~~~~~~~~~~~~~~~~
>       uamqp/c_uamqp.c:111209:3: warning: unused label 'bad' [-Wunused-label]
>         bad:
>         ^~~~
>       35 warnings and 1 error generated.
>       error: command '/usr/bin/clang' failed with exit code 1
>       [end of output]
> ```

Did you ever get an answer?

--- a/requirements.txt
+++ b/requirements.txt
@@ -5,7 +5,7 @@ azure-containerregistry==1.2.0
 azure-core==1.38.0
 azure-eventhub==5.15.0
 azure-identity==1.19.0
-azure-iot-hub==2.6.1
+azure-iot-hub==2.6.1;platform_machine=="x86_64"
 azure-keyvault==4.2.0
 azure-keyvault-securitydomain==1.0.0b1
 azure-mgmt-apimanagement==4.0.1

